controllers/djob_controller_databricks.go

```diff
@@ -88,13 +92,44 @@ func (r *DjobReconciler) delete(instance *databricksv1alpha1.Djob) error {
 
 	jobID := instance.Status.JobStatus.JobID
 
+	_, err := getJob(r, jobID)
+
 	// Check if the job exists before trying to delete it
-	if _, err := r.APIClient.Jobs().Get(jobID); err != nil {
+	if err != nil {
 		if strings.Contains(err.Error(), "does not exist") {
 			return nil
 		}
 		return err
 	}
 
-	return r.APIClient.Jobs().Delete(jobID)
+	return trackExecutionTime(djobDeleteDuration, func() error {
+		return r.APIClient.Jobs().Delete(jobID)
+	})
+}
+
+func getJob(r *DjobReconciler, jobID int64) (job models.Job, err error) {
+	return trackJob(djobGetDuration, djobGetSuccess, djobGetFailure, func() (models.Job, error) {
+		return r.APIClient.Jobs().Get(jobID)
+	})
+}
+
+func createJob(r *DjobReconciler, instance *databricksv1alpha1.Djob) (job models.Job, err error) {
+	return trackJob(djobCreateDuration, djobCreateSuccess, djobCreateFailure, func() (models.Job, error) {
+		return r.APIClient.Jobs().Create(*instance.Spec)
+	})
+}
+
+func trackJob(duration prometheus.Histogram, success prometheus.Counter, failure prometheus.Counter, 
+	f func() (models.Job, error)) (job models.Job, err error) {
+		job, err = trackJobExecutionTime(duration, func() (models.Job, error) {
```

Comment:
personal: I'd maybe roll the `trackJobExecutionTime` method and `trackExecutionTime` up into this method to keep things clearer but I'm not dead certain on that.

Reply:
You could use the `defer` pattern to track the start and end time of the func.

Reply:
I agree `trackJobExecutionTime` should be rolled up, but the `trackExecutionTime` is more generic and designed for use with other types other than job